File: backend/domain/stock/garment.go
package stock

import "stokkit/domain/base"

type Garment struct {
	base.Entity
	GarmentValues
}

type GarmentValues struct {
	Year        int    `json:"year"`
	Style       string `json:"style"`
	Description string `json:"description"`
}

func (g *Garment) SetKey(id int) {
	g.ID = id
}

func (g *Garment) GetKey() int {
	return g.ID
}

File: frontend/src/domain/stock/garment.ts
import { Entity } from "../base/entity";

export class GarmentValues {
	public year: number;
	public description: string;
	public style: string;

	constructor(year: number, description: string, style: string) {
		this.year = year;
		this.description = description;
		this.style = style;
	}
}

// A "Garment" is both a set of GarmentValues
// and persistence details (or an Entity).
export type Garment = GarmentValues & Entity